js/main.js

```diff
@@ -18,16 +19,42 @@ function submitForm(event) {
     notes: formElements.elements.notes.value,
     entryId: data.nextEntryId,
   };
-  data.nextEntryId++;
-  data.entries.unshift(formEntries);
-  imgElement.src = '../images/placeholder-image-square.jpg';
-  form.reset();
-  const unorderedList = document.querySelector('ul');
-  unorderedList.appendChild(renderEntry(formEntries));
-  viewSwap('entries');
-  if (localStorage.getItem('form-submission') === null) {
-    toggleNoEntries();
+
+  if (data.editing === null) {
+    data.nextEntryId++;
+    data.entries.unshift(formEntries);
+    imgElement.src = '../images/placeholder-image-square.jpg';
+    form.reset();
+    const unorderedList = document.querySelector('ul');
+    unorderedList.appendChild(renderEntry(formEntries));
+    viewSwap('entries');
+    if (localStorage.getItem('form-submission') === null) {
+      toggleNoEntries();
+    }
   }
+  if (data.editing !== null) {
+    formEntries.entryId = data.editing.entryId;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        data.entries[i] = formEntries;
+      }
+    }
+
+    const currentLiElements = document.querySelectorAll('li');
+    for (let i = 0; i < currentLiElements.length; i++) {
+      if (
+        currentLiElements[i].getAttribute('data-entry-id') ==
+        data.editing.entryId
+      ) {
+        currentLiElements[i] = renderEntry(formEntries);
+      }
+    }
+
+    document.querySelector('h2').textContent = 'New Entry';
+    data.editing = null;
+    form.reset();
+  }
+  location.reload();
```

Comment:
you can delete this line once you fix line 29 - I think you were doing this to force the saved list of entries to reorder in the proper order. Doing this causes a weird user experience when the list renders for split second in the wrong order, the page refreshes and the list is in the opposite order.